porch/pkg/git/git.go

```diff
@@ -382,6 +382,14 @@ func (r *gitRepository) DeletePackageRevision(ctx context.Context, old repositor
 		// Delete the tag
 		refSpecs.AddRefToDelete(ref)
 
+		// If this revision was proposed for deletion, we need to delete the associated branch.
+		refSpecsForDeletionProposed := newPushRefSpecBuilder()
+		deletionProposedBranch := createDeletionProposedName(oldGit.path, oldGit.revision)
+		refSpecsForDeletionProposed.AddRefToDelete(plumbing.NewHashReference(deletionProposedBranch.RefInLocal(), oldGit.commit))
+		if err := r.pushAndCleanup(ctx, refSpecsForDeletionProposed); err != nil {
+			// the deletionProposed branch might not have existed, so we ignore any errors here
```

Comment:
Should we ignore all errors? Seems like we probably should try to detect a NotFound error and propagate any other errors. Maybe add a todo for that.

Reply:
I looked into it a little and I think this is the error that gets thrown if the branch doesn't exist: https://github.com/go-git/go-git/blob/f0b111ab70e4e90013658b0835929b2083902017/remote.go#L1411

I updated it so that we specifically check for this error with string comparisons. I can update it or add a TODO if you have some more feedback on this.